instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteAlternativesTest.kt

```diff
@@ -40,19 +43,25 @@ class RouteAlternativesTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::cla
     @get:Rule
     val mockLocationReplayerRule = MockLocationReplayerRule(mockLocationUpdatesRule)
 
+    @get:Rule
+    val mapboxHistoryTestRule = MapboxHistoryTestRule()
+
     private lateinit var mapboxNavigation: MapboxNavigation
 
     @Before
     fun setup() {
-        val routeAlternativesOptions = RouteAlternativesOptions.Builder()
-            .intervalMillis(TimeUnit.SECONDS.toMillis(10))
-            .build()
-        mapboxNavigation = MapboxNavigationProvider.create(
-            NavigationOptions.Builder(activity)
-                .accessToken(getMapboxAccessTokenFromResources(activity))
-                .routeAlternativesOptions(routeAlternativesOptions)
+        runOnMainSync {
```

Comment:
is this the source of the issue? not running the MapboxNavigationProvider.create on the main sync?

Reply:
Yeah one of them .On instrumentation tests we need to run the `MapboxNavigation` creation on the main thread, if not the navigator observer is not set and all the interactions with the `Navigator` are lost (NN added that "recently"). The were a couple more. 1, had to relax expected coordinates to handle MockWebServer re-routes (as we don't know exactly when NN is going to emit the `OFF_ROUTE`) and 2, the NN 32-bit / fallback mode 💥